### PR TITLE
[draft][circle-quantizer] Support OneHot quantization

### PR DESCRIPTION
**Pull request by m-bronnikov**
This draft enables quantization of models with OneHot ops.

Signed-off-by: Maksim Bronnikov <max120199@gmail.com>

----------------

For: #8065
+cc @jinevening

### CODE REVIEW

**Comment by m-bronnikov** [comment]
This draft resolves #8065 (Model from "What?" quantizes well).

File: compiler/record-minmax/src/MinMaxObserver.cpp [hunk 1]
```diff
@@ -51,6 +51,16 @@ void MinMaxObserver::postTensorWrite(const luci::CircleNode *node,
...
+  if (node->dtype() == DataType::S32)
+  {
+    // Integer type tensor is not quantized
+    return;
+  }
+  if (node->dtype() == DataType::S64)
+  {
+    // Integer type tensor is not quantized
+    return;
+  }
```
Comment (binarman) [inline compiler/record-minmax/src/MinMaxObserver.cpp:63]: I see that this code is already merged, just want to ask how this code is interacts with code below?
```
if (tensor->element_type() != DataType::FLOAT32)
```

Comment (m-bronnikov) [reply compiler/record-minmax/src/MinMaxObserver.cpp:63]: Personally, I think it's OK to skip quantization for all integer tensors and just return, when observer detects node with integer type. But as I understand now, this code makes `switch`/`case` statement bellow redundant in most cases. So, I think better to ask some advice about correctness of this approach.

@jinevening Could you please take a look? =)

Comment (binarman) [reply compiler/record-minmax/src/MinMaxObserver.cpp:63]: > But as I understand now, this code makes switch/case statement bellow redundant in most cases

Yes, this is what attracted my attention =)

Comment (jinevening) [reply compiler/record-minmax/src/MinMaxObserver.cpp:63]: record-minmax records min/max of fp32 feature maps. AFAIK, int32 and int64 are not used for feature maps, so I approved the change (@binarman please tell me if I missed something).

It seems that the case for ARG_MAX becomes redundant in the below switch statement. @m-bronnikov could you remove the case?

**Comment by m-bronnikov** [comment]
I think this draft could be closed as already merged.

File: compiler/record-minmax/src/MinMaxObserver.cpp (hunk 1 above)
Comment (binarman) [reply compiler/record-minmax/src/MinMaxObserver.cpp:63]: > int32 and int64 are not used for feature maps

So far I saw integer operations only for indexing, so you are right.

**Pull request closed without merging**